### PR TITLE
Start documenting review process

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: docs/dev/README.md [hunk 1]
```diff
@@ -117,6 +117,113 @@ Additionally, I use `cargo run --release -p rust-analyzer -- analysis-stats
...
+   Specifically, you add a `pub use` reexport from another crate or you add a new line to `[dependencies]` section of `Cargo.toml`.
+   A good example here would be adding reference search capability to the assists crates.
+
+For the first group, the change is generally merged as long as:
+
+* it works for the happy case,
+* it has tests,
+* it doesn't panic for unhappy case.
+
+For the second group, the change would be subjected to a quite a bit of scrutiny and iteration.
```
Comment (kjeremy) [inline docs/dev/README.md:156]:
```suggestion
For the second group, the change would be subjected to quite a bit of scrutiny and iteration.
```

File: docs/dev/README.md [hunk 2]
```diff
@@ -117,6 +117,113 @@ Additionally, I use `cargo run --release -p rust-analyzer -- analysis-stats
...
+The new API needs to be right (or at least easy to change later).
+The actual implementation doesn't matter that much.
+It's very important to minimize the amount of changed lines of code for changes of the second kind.
+Often, you start doing change of the first kind, only to realise that you need to elevate to a change of the second kind.
+In this case, we'll probably ask you to split API changes into a separate PR.
+
+Changes of the third group should be pretty rare, so we don't specify any specific process for them.
+That said, adding an innocent-looking `pub use` is a very simple way to break encapsulation, keep and eye on it!
```
Comment (kjeremy) [inline docs/dev/README.md:164]:
```suggestion
That said, adding an innocent-looking `pub use` is a very simple way to break encapsulation, keep an eye on it!
```

**Comment by kjeremy** [comment]
As small as it is I think mentioning `cargo xtask format` would be a good idea since it takes the burden of formatting off of the developer.

**Comment by matklad** [comment]
>As small as it is I think mentioning cargo xtask format would be a good idea since it takes the burden of formatting off of the developer.

We mention formatting checks on CI and `pre-commit-hook`. I am not sure if `xtask format` is all that useful today, I just use `cargo fmt` myself. 

bors r+

**Pull request merged**